### PR TITLE
Update cloudbuild.yaml to use Go modules

### PR DESCRIPTION
**Pull request by navierula**

### CODE REVIEW

**Review by nityaravi — APPROVED** [review]
LGTM pending Jason's approval.

File: cloudbuild.yaml
Comment (imjasonh) [inline cloudbuild.yaml:3]: Go mod init should only be necessary once, some time before this build is repeatedly run.

File: cloudbuild.yaml
Comment (imjasonh) [inline cloudbuild.yaml]: This env is set already inside the golang image, so this isn't necessary.

**Review by imjasonh — CHANGES_REQUESTED** [review]

File: cloudbuild.yaml
Comment (navierula) [reply cloudbuild.yaml:3]: Added a comment about this.

File: cloudbuild.yaml
Comment (navierula) [reply cloudbuild.yaml]: removed.

**Review by navierula — COMMENTED** [review]
made changes.

File: cloudbuild.yaml
Comment (imjasonh) [reply cloudbuild.yaml:3]: I still think this should be a one-time-only step documented _outside_ the cloudbuild.yaml file.

The reason being that most users are just going to blindly copypaste whatever's in here and ignore the comments (I know I would). 

The only users who should have to do this are authors of packages since before modules, who haven't migrated to modules yet, which should hopefully be a small and shrinking set of the population.

Comment (nityaravi) [reply cloudbuild.yaml:3]: Documenting this outside this cloudbuild.yaml sgtm. i think we can add these instructions in the README file.

**Review by imjasonh — APPROVED** [review]

**Pull request merged**
